engine/execution/computation/computer/computer.go

```diff
@@ -96,111 +106,126 @@ func (e *blockComputer) ExecuteBlock(
 }
 
 func (e *blockComputer) executeBlock(
-	ctx context.Context,
+	blockSpan opentracing.Span,
 	block *entity.ExecutableBlock,
 	stateView state.View,
 	programs *programs.Programs,
 ) (*execution.ComputationResult, error) {
 
 	blockCtx := fvm.NewContextFromParent(e.vmCtx, fvm.WithBlockHeader(block.Block.Header))
-
 	collections := block.Collections()
-
-	var gasUsed uint64
-
-	interactions := make([]*delta.SpockSnapshot, len(collections)+1)
-
-	events := make([]flow.Event, 0)
-	serviceEvents := make([]flow.Event, 0)
-	blockTxResults := make([]flow.TransactionResult, 0)
+	res := &execution.ComputationResult{
+		ExecutableBlock:    block,
+		Events:             make([]flow.Event, 0),
+		ServiceEvents:      make([]flow.Event, 0),
+		TransactionResults: make([]flow.TransactionResult, 0),
+		StateCommitments:   make([]flow.StateCommitment, 0),
+		Proofs:             make([][]byte, 0),
+	}
 
 	var txIndex uint32
+	var err error
+	var wg sync.WaitGroup
+
+	stateCommitments := make([]flow.StateCommitment, 0, len(collections)+1)
+	proofs := make([][]byte, 0, len(collections)+1)
+
+	bc := blockCommitter{
```

Comment:
I think we can simplify the synchronization here a bit.

We have the wait group, which we increment for each chunk, then we have the `views` channel, which we send views to commit over, then close. By closing the channel, we will exit [the `range` loop here](https://github.com/onflow/flow-go/pull/585/files#diff-94b6f84058d4bc409d139e005e5e484de3d76dab187c7c4f71292178dafd8e76R354). All we need from the wait group is to let us know when `Run` goroutine exits, after which all the views have necessarily been committed. So I think we can skip "counting" chunks with the wait group and just do:
```go
var wg sync.Waitgroup
wg.Add(1)
go func() {
    bc.Run()
    wg.Done()
}()
```

then after all the calls to `Commit`:
```go
close(bc.views)
wg.Wait()
```

Reply:
yeah that sounds cleaner, changed at fd893bd

Reply:
Will closing wait for all items to be processed? In other words - what happens if we are closing before all items are processed? Or when not items are pushed yet (ie. execution of last chunk takes longer?)

Reply:
seems to be safe https://play.golang.org/p/NWKu4quASaZ

Reply:
I don't think we have a case that items are not pushed.

Reply:
We have to push all the items into the channel before closing it, but it's safe to close the channel while there are items in the channel or if the goroutine reading from the channel hasn't read/processed everything from the channel. We should be OK here since there's just one writer goroutine (the main one) and we close the channel after the last `bc.Commit` call.

Reply:
OK, good!